pkg/instancetype/instancetype.go

```diff
@@ -454,26 +426,23 @@ func (m *methods) FindInstancetypeSpec(vm *virtv1.VirtualMachine) (*instancetype
 }
 
 func (m *methods) getInstancetypeSpecRevision(revisionName string, namespace string) (*instancetypev1alpha1.VirtualMachineInstancetypeSpec, error) {
-
 	revision, err := m.clientset.AppsV1().ControllerRevisions(namespace).Get(context.Background(), revisionName, metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
 
-	instancetypeSpecRevision := instancetypev1alpha1.VirtualMachineInstancetypeSpecRevision{}
-	err = json.Unmarshal(revision.Data.Raw, &instancetypeSpecRevision)
-	if err != nil {
+	if err := decodeRevisionObject(revision); err != nil {
 		return nil, err
 	}
 
-	// For now we only support a single version of VirtualMachineInstancetypeSpec but in the future we will need to handle older versions here
-	instancetypeSpec := instancetypev1alpha1.VirtualMachineInstancetypeSpec{}
-	err = json.Unmarshal(instancetypeSpecRevision.Spec, &instancetypeSpec)
-	if err != nil {
-		return nil, err
+	switch obj := revision.Data.Object.(type) {
+	case *instancetypev1alpha1.VirtualMachineInstancetype:
```

Comment:
Is there a reason we can't collapse these cases into one?

Reply:
Yes, it is not possible in golang type system. Which type would the `obj` variable have in the combined branch?
It would need to be some common interface, but then we could not access `.Spec` field.

Reply:
Ah apologies I see `obj` stays as `runtime.Object` without access to `.Spec`, I just recalled that you can match on multiple types and didn't think about access to `.Spec` via `obj` here.